src/test/unit/azure/azureLogin/azureAuth.test.ts

```diff
@@ -0,0 +1,211 @@
+            succeeded: false,
+            error: "failed"
+        };
+        azureSessionProvider.getAuthSession = function (options?: GetAuthSessionOptions | undefined): Promise<GeneralUtils.Errorable<AzureAuthenticationSession>> {
+            return Promise.resolve(failedSession);
+        };
+        sandbox.stub(AzureAuth, 'isReady').returns(false);
+        let getSubscriptionsStub = sandbox.stub(AzureSessionProviderHelper, "getSessionProvider").returns(azureSessionProvider);
+        const res = await AzureAuth.getReadySessionProvider();
+        sandbox.assert.calledOnce(getSubscriptionsStub);
+        assert.equal(res.succeeded, false);
+        assert.equal((res as GeneralUtils.Failed).error, "Failed to get authentication session: failed");
+    });
+});
+
+describe("getConfiguredAzureEnv", () => {
+    let sandbox = null as any;
+    let workspaceConfiguration: vscode.WorkspaceConfiguration;
+    before(() => {
+        sandbox = sinon.createSandbox();
+    });
+    beforeEach(() => {
+        workspaceConfiguration = {
+            get: function <T>(section: string): T | undefined {
+                throw new Error("Function not implemented.");
+            },
+            has: function (section: string): boolean {
+                throw new Error("Function not implemented.");
+            },
+            inspect: function <T>(section: string): { key: string; defaultValue?: T | undefined; globalValue?: T | undefined; workspaceValue?: T | undefined; workspaceFolderValue?: T | undefined; defaultLanguageValue?: T | undefined; globalLanguageValue?: T | undefined; workspaceLanguageValue?: T | undefined; workspaceFolderLanguageValue?: T | undefined; languageIds?: string[] | undefined; } | undefined {
+                throw new Error("Function not implemented.");
+            },
+            update: function (section: string, value: any, configurationTarget?: boolean | vscode.ConfigurationTarget | null | undefined, overrideInLanguage?: boolean | undefined): Thenable<void> {
+                throw new Error("Function not implemented.");
+            }
+        };
+    });
+    afterEach(() => {
+        sandbox.restore();
+    });
+    it('getConfiguredAzureEnv return azure china', async () => {
+        sandbox.stub(workspaceConfiguration, "get").returns('ChinaCloud');
+        sandbox.stub(vscode.workspace, "getConfiguration").returns(workspaceConfiguration);
+        const res = AzureAuth.getConfiguredAzureEnv();
+        assert.strictEqual(res.name, "AzureChinaCloud");
+    });
+    it('getConfiguredAzureEnv return USGovernment', async () => {
+        sandbox.stub(workspaceConfiguration, "get").returns('USGovernment');
+        sandbox.stub(vscode.workspace, "getConfiguration").returns(workspaceConfiguration);
+        const res = AzureAuth.getConfiguredAzureEnv();
+        assert.strictEqual(res.name, "AzureUSGovernment");
+    });
+    it('getConfiguredAzureEnv return USGovernment', async () => {
```

Comment:
Not 'getConfiguredAzureEnv return USGovernment'?

Reply:
updadted.